src/Microsoft.OData.Core/UriParser/Aggregation/AggregationMethod.cs

```diff
@@ -9,7 +9,31 @@ namespace Microsoft.OData.UriParser.Aggregation
     /// <summary>
     /// Enumeration of methods used in the aggregation clause
     /// </summary>
-    public enum AggregationMethod
+    public sealed class AggregationMethod
+    {
+        public AggregationMethodType MethodType { get; private set; }
+        public string MethodLabel { get; private set; }
+
+        private AggregationMethod(AggregationMethodType aggregationMethodType)
+        {
+            this.MethodType = aggregationMethodType;
+        }
+
+        public static AggregationMethod Sum = new AggregationMethod(AggregationMethodType.Sum);
+        public static AggregationMethod Min = new AggregationMethod(AggregationMethodType.Min);
+        public static AggregationMethod Max = new AggregationMethod(AggregationMethodType.Max);
+        public static AggregationMethod Average = new AggregationMethod(AggregationMethodType.Average);
+        public static AggregationMethod CountDistinct = new AggregationMethod(AggregationMethodType.CountDistinct);
+
+        public static AggregationMethod Custom(string customMethodLabel)
+        {
+            var aggregationMethod = new AggregationMethod(AggregationMethodType.Custom);
+            aggregationMethod.MethodLabel = customMethodLabel;

```

Comment:
I was asserting it just before calling the constructor, in order to throw the right parse error. But it's a good idea to do it in the constructor too. Also added a comment with the assert reason.